start_vscode_bessemer.sh

```diff
@@ -0,0 +1,478 @@
+#                                                                             #
+#  Main author    : Samuel Fux                                                #
+#  Contributions  : Andreas Lugmayr                                           #
+#  Date           : October 2021                                              #
+#  Location       : ETH Zurich                                                #
+#  Version        : 0.1                                                       #
+#  Change history :                                                           #
+#                                                                             #
+#  28.10.2021    Initial version of the script based on Jupyter script        #
+#  Forked and edited for TUoS by J.Moore                                      #
+###############################################################################
+
+###############################################################################
+# Configuration options, initalising variables and setting default values     #
+###############################################################################
+
+# Version
+VSC_VERSION="0.1"
+
+# Script directory
+VSC_SCRIPTDIR=$(pwd)
+
+# hostname of the cluster to connect to
+VSC_HOSTNAME="bessemer.shef.ac.uk"
+
+# order for initializing configuration options
+# 1. Defaults values set inside this script
+# 2. Command line options overwrite defaults
+# 3. Config file options  overwrite command line options
+
+# Configuration file default    : $HOME/.vsc_config
+VSC_CONFIG_FILE="$HOME/.vsc_config"
+
+# Username default              : no default
+VSC_USERNAME=""
+
+# Number of cpus per tasks default : 1 task per node
+VSC_CPUS_PER_TASK=1
+
+# Runtime limit default         : 1:00 hour
+VSC_RUN_TIME="01:00:00"
+
+# Memory default                : 2 GB per job
+VSC_MEM_PER_JOB=2
+
+# Number of GPUs default        : 0 GPUs
+VSC_NUM_GPU=0
+
+# Partition ID default        : gpu
+VSC_PARTITION_ID="gpu"
+
+# Waiting interval default      : 30 seconds
+VSC_WAITING_INTERVAL=30
+
+# SSH key location default      : no default
+VSC_SSH_KEY_PATH=""
+
+###############################################################################
+# Usage instructions                                                          #
+###############################################################################
+
+function display_help {
+cat <<-EOF
+$0: Script to start a VSCode remote server on Bessemer from a local computer
+
+Usage: start_vscode_bessemer.sh [options]
+
+Options:
+
+        -u | --username       USERNAME                  TUoS username for SSH connection to Bessemer
+        -W | --runtime        RUN_TIME                  Run time limit for the code-server in hours and minutes HH:MM
+        -n | --numcpus        NUM_CPUS_PER_TASK         Number of CPU cores per task     
+        -m | --memory         MEM_PER_JOB               Memory limit in GB per job. (RAM) Ex. 4 cores *4G = 16 
```

Comment:
This needs amending to Bessemer parlance (memory per node.)

Reply:
Adjusting the `VSC_MEM_PER_JOB` variable to `VSC_MEM_PER_NODE` may also be wise.

Reply:
Thank you. Edited